docs/developer-guide/ksqldb-rest-api/streaming-endpoint.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: page
+title: HTTP Streaming API
+tagline: streaming endpoints
+description: The HTTP Streaming API lets you execute pull or push queries and stream inserts to the
+server
+keywords: ksqlDB, query, insert, select
+---
+
+Please note: These endpoints are used by the ksqlDB Java client. If you are using Java you might want
+to consider using the Java client rather than using this API directly.
+
+These endpoints are only available when using HTTP 2.
+
+### Executing pull or push queries
+
+The request method will be a POST.
+
+Requests will be made to the endpoint "/query-stream".
+
+The body of the request is a JSON object UTF-8 encoded as text, containing the arguments for the
+operation (newlines have been added here for the sake of clarity but the real JSON must not contain
+ unescaped newlines)
+
+````
+{
+"sql": "select * from foo", <----- the SQL of the query to execute
+"properties": {             <----- Optional properties for the query
+    "prop1": "val1",
+    "prop2": "val2"
+   }
+}
+
+````
+
+The endpoint produces responses with two possible content types `application/json` and
+`application/vnd.ksqlapi.delimited.v1`. To choose one or the other you should set the `Accept`
+header in the request. The default is `application/vnd.ksqlapi.delimited.v1`.
+
+In the case of a successful query, if the content type is `application/vnd.ksqlapi.delimited.v1`
+then the results are returned as a header JSON object followed by zero or more JSON arrays
+delimited by newlines. Newline delimited formats are easy to parse by clients, and don't require
+a streaming JSON parser on the client in the case that intermediate results need to be output.
+
+````
+{
+"queryId", "xyz123",                          <---- unique ID of the query, used when terminating the query

```

Comment:
Clarify that this is empty in the case of pull queries (and non-empty in the case of push queries).